tmva/tmva/src/ResultsMulticlass.cxx

```diff
@@ -85,32 +85,38 @@ Double_t TMVA::ResultsMulticlass::EstimatorFunction( std::vector<Double_t> & cut
 
    DataSet* ds = GetDataSet();
    ds->SetCurrentType( GetTreeType() );
-   Float_t truePositive = 0;
-   Float_t falsePositive = 0;
-   Float_t sumWeights = 0;
 
-   for (Int_t ievt=0; ievt<ds->GetNEvents(); ievt++) {
-      const Event* ev = ds->GetEvent(ievt);
-      Float_t w = ev->GetWeight();
-      if(ev->GetClass()==fClassToOptimize)
-         sumWeights += w;
-      bool passed = true;
-      for(UInt_t icls = 0; icls<cutvalues.size(); ++icls){
-         if(cutvalues.at(icls)<0. ? -fMultiClassValues[ievt][icls]<cutvalues.at(icls) : fMultiClassValues[ievt][icls]<=cutvalues.at(icls)){
-            passed = false;
+   // Cache optimisation, count true and false positives with memory access
+   // instead of code branch.
+   Float_t positives[2] = {0, 0};
+
+   for (Int_t ievt = 0; ievt < ds->GetNEvents(); ievt++) {
+      UInt_t  evClass = fEventClasses[ievt];
+      Float_t w       = fEventWeights[ievt];
+
+      Bool_t break_outer_loop = false;
+      for (UInt_t icls = 0; icls < cutvalues.size(); ++icls) {
```

Comment:
Just a suggestion: In order to avoid break/continue, how about refactoring this for block to its own function that returns a boolean true/false where true would instead of continue on line 108, run the two other lines (111, 112)?